drivers/ad5770r/ad5770r.c

```diff
@@ -0,0 +1,650 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "platform_drivers.h"
+#include "ad5770r.h"
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+/**
+ * Read from device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_read(struct ad5770r_dev *dev,
+			     uint8_t reg_addr,
+			     uint8_t *reg_data)
+{
+	uint8_t buf[3];
+	int32_t ret;
+
+	buf[0] = AD5770R_REG_READ(reg_addr);
+	buf[1] = 0x00;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, 2);
+	*reg_data = buf[1];
+
+	return ret;
+}
+
+/**
+ * Multibyte read from device. A register read begins with the address
+ * and autoincrements for each aditional byte in the transfer.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @param count - Number of bytes to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_read_multiple(struct ad5770r_dev *dev,
+				      uint8_t reg_addr,
+				      uint8_t *reg_data,
+				      uint16_t count)
+{
+	uint8_t buf[1024];
+	uint16_t i;
+	int32_t ret;
+
+	/* Check if multibyte enabled */
+
+	buf[0] = AD5770R_REG_READ(reg_addr);
+	memset(&buf[1], 0x00, count);
+
+	ret = spi_write_and_read(dev->spi_desc, buf, count + 1);
+
+	for (i = 0; i < count; i++)
+		reg_data[i] = buf[i+1];
+
+	return ret;
+}
+/**
+ * Write to device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_write(struct ad5770r_dev *dev,
+			      uint8_t reg_addr,
+			      uint8_t reg_data)
+{
+	int32_t ret;
+	uint8_t buf[2];
+
+	buf[0] = AD5770R_REG_WRITE(reg_addr);
+	buf[1] = reg_data & 0xFF;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, 2);
+
+	return ret;
+}
+
+/**
+ * Multibyte write from device. A register write begins with the address
+ * and autoincrements for each additional byte in the transfer.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @param count - Number of bytes to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_write_multiple(struct ad5770r_dev *dev,
+				       uint8_t reg_addr,
+				       uint8_t *reg_data,
+				       uint16_t count)
+{
+	int32_t ret;
+	uint8_t data[1024];
+
+	data[0] = AD5770R_REG_WRITE(reg_addr);
+	memcpy(&data[1], reg_data, count);
+
+	ret = spi_write_and_read(dev->spi_desc, data, count + 1);
```

Comment:
you could do `return spi_write_and_read(dev->spi_desc, data, count + 1);`   here directly;